src/scripts/main.js

```diff
@@ -1 +1,82 @@
 'use strict';
+
+const doc = document.querySelector('body');
+
+let leftClick = false;
+let rightClick = false;
+let firstPromiseRejected = false;
+
+const createDiv = (classN, textC) => {
+  const div = document.createElement('div');
+  div.setAttribute('data-qa', 'notification');
+  div.className = classN;
+  div.textContent = textC;
+  doc.appendChild(div);
+};
+const firstPromise = new Promise((resolve, reject) => {
+  const rejectTimer = setTimeout(() => {
+    createDiv('error', 'First promise was rejected');
+    firstPromiseRejected = true;
+    reject();
+  }, 3000);
+
+  doc.addEventListener('click', (e) => {
+    if (e.button === 0 && !firstPromiseRejected) {
+      clearTimeout(rejectTimer);
+      resolve();
+    }
+  });
+});
+
+firstPromise
+  .then(() => {
+    if (!firstPromiseRejected) {
+      createDiv('success', 'First promise was resolved');
+    }
+  })
+  .catch(() => {});
+
+const secondPromise = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (!firstPromiseRejected) {
```

Comment:
why do we need all this logic with firstPromiseRejected, looks redundant for me

Reply:
I added this logic so that when pressing "for example" -  the right mouse button, if the left mouse button had already been pressed, the message from the second promise would not be displayed again and this message would not be overlapped by the third message from the last promise